src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java

```diff
@@ -30,13 +30,12 @@ public static <T> ResponseWrapper<T> success(T payload) {
         return success(payload, null);
     }
 
-    @Deprecated
-    public static <T> ResponseWrapper<T> failure(T payload, String message) {
-        return new ResponseWrapper<>(null, false, payload, message);
+    public static <T> ResponseWrapper<T> failure(List<ErrorResponse> errors) {
+        return failure(errors, null);
     }
 
-    public static <T> ResponseWrapper<T> failure(List<ErrorResponse> errors) {
-        return new ResponseWrapper<>(errors, false, null, null);
+    public static <T> ResponseWrapper<T> failure(List<ErrorResponse> errors, T payload) {
+        return new ResponseWrapper<>(errors, false, payload, null);
```

Comment:
So for failure(), errors have to be there and payload is optional now.

We cannot go about not having payload because of one specific use case. If we change the requirement, then there is a way to get rid of this method altogether.